flytekit/image_spec/image_spec.py

```diff
@@ -163,7 +181,7 @@ def exist(self) -> Optional[bool]:
             return None
 
     def __hash__(self):
-        return hash(asdict(self).__str__())
+        return hash(self.id)
```

Comment:
Can this just return `self.id`?

Reply:
it should return a number instead of string